src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -212,9 +213,14 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
                 }
 
                 PublishCode();
+                codeGotPublished = true;
             }
             finally
             {
+                if (!codeGotPublished)
+                {
+                    PublishEmptyCode();
```

Comment:
This needs to be under an ifdef - it breaks CoreRT's ability to recompile the method with a throwing method body (we are only allowed to call `SetCode` once).